app/src/main/java/com/eis0/library_demo/PollManager.java

```diff
@@ -0,0 +1,205 @@
+ * Right now it communicates directly with SMSManager, but in the future it will send messages to a
+ * class handling messages longer than a single SMS.
+ * @author Giovanni Velludo
+ * @author Matteo Carnelos
+ */
+public class PollManager implements ReceivedMessageListener<SMSMessage> {
+
+    // NOTE: FIELD_SEPARATOR is a regex, there are some illegal values (e.g. "*")
+    // must not be private for tests to work, not a big deal since it's final
+    // TODO: Find a stronger FIELD_SEPARATOR (e.g. Escape Char)
+    static final String FIELD_SEPARATOR = "\r";
+    static final String NEW_POLL_MSG_CODE = "0";
+    static final String ANSWER_MSG_CODE = "1";
+    private static final String YES_ANSWER_CODE = "1";
+    private static final String NO_ANSWER_CODE = "0";
+
+    // Must always be static, because PollManager is a singleton
+    private static PollManager instance = null;
+    private PollListener pollListener;
+    private SMSManager smsManager = SMSManager.getInstance();
+
+    private static HashMap<Integer, TernaryPoll> sentPolls = new HashMap<>();
+
+    /**
+     * PollManager constructor, sets this as the SMSManager listener.
+     * It cannot be accessed from outside the class because this follows the Singleton Design Pattern.
+     * @author Giovanni Velludo
+     * @author Matteo Carnelos
+     */
+    private PollManager() {
+        smsManager.addReceiveListener(this);
+    }
+
+    /**
+     * Returns a new instance of PollManager if none exist, otherwise the one already created as per
+     * the Singleton Design Patter.
+     * @return The only instance of this class.
+     * @author Giovanni Velludo
+     */
+    public static PollManager getInstance() {
+        if(instance == null) instance = new PollManager();
+        return instance;
+    }
+
+    /**
+     * Set the listener that will be called when a new poll or an answer is received.
+     * @param listener The listener to wake up. It must implement the PollListener interface.
+     * @author Giovanni Velludo
+     * @author Matteo Carnelos
+     */
+    void setPollListener(PollListener listener) {
+        pollListener = listener;
+    }
+
+    /**
+     * Creates a new poll and sends it to all the included users, it eventually wakes the listener.
+     * @param name The name given to the poll.
+     * @param question The question to ask users.
+     * @param users Users to which the question should be asked. They must be valid phone numbers.
+     * @throws IllegalArgumentException When users is empty
+     * @author Giovanni Velludo
+     * @author Matteo Carnelos
+     */
+    void createPoll(String name, String question, ArrayList<SMSPeer> users)
+            throws  IllegalArgumentException {
+        if (users.isEmpty()) throw new IllegalArgumentException("Can't create poll with no users");
+        TernaryPoll poll = new TernaryPoll(name, question, users);
+        sentPolls.put(poll.getPollId(), poll);
```

Comment:
I suppose TernaryPoll is a class that you or your group created, a question about poll.getPollId():
Does it return a unique id? How do you manage that?

Reply:
Yes, if you want to check it out you can take a look at TernaryPoll.java in code_review/giovannivelludo_pull_request. TernaryPoll has a private static field named pollCount which is incremented each time a new poll is created and is used as that poll's id.

Reply:
Nice, thanks

Reply:
Scusa, parlo in italiano perchè voglio annotare velocemente questa cosa e non vorrei farla pesare sulla review del tuo codice, ma mi è venuto in mente questa cosa durante la giornata:
Avendo quel campo pollCount statico, per tutta l'esecuzione del codice l'area di memoria che contiene il campo viene condivisa tra le varie istanze e ok. Ma tra le varie esecuzioni il valore si perde. Allora mi sembra che ci possano essere dei conflitti tra le varie esecuzioni dell'app. Avete tenuto conto della cosa salvando da qualche parte oppure per il mandante dei sondaggi, questi esistono solo finché tiene aperta l'app? Spulciando qua e la non mi è parso.
Il tuo codice mi piace molto devo dire, pulitissimo.

Reply:
In the first version of the app everything is reset when the app is destroyed; my group mate who's working on saving and restoring data from disk merged his changes to the main branch yesterday, but I didn't merge them to my branch because I wanted to avoid messing up my code right before the review.

Reply:
Ok, giusto una curiosità 😊